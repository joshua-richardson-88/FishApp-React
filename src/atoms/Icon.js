import React from 'react';

export default function Icon(props) {
  return (
    <span
      className='fish-icon'
      role='img'
      aria-label={props.ariaLabel}
      onMouseEnter={props.onMouseEnter}
      onMouseLeave={props.onMouseLeave}
      data-testid='icon'
    >
      {props.type}
    </span>
  );
}
