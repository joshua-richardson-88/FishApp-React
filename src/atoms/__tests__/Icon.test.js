import React from 'react';
import Icon from '../Icon';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Atoms - Icon Test', () => {
  const mIn = jest.fn();
  const mOut = jest.fn();

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <Icon type='foo' ariaLabel='a' onMouseEnter={mIn} onMouseLeave={mOut} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders icon text', () => {
    const { getByTestId } = render(
      <Icon type='foo' ariaLabel='a' onMouseEnter={mIn} onMouseLeave={mOut} />
    );

    expect(getByTestId('icon')).toHaveTextContent(/^foo$/);
  });
  it('renders icon aria-label', () => {
    const { getByRole } = render(
      <Icon type='foo' ariaLabel='a' onMouseEnter={mIn} onMouseLeave={mOut} />
    );

    expect(getByRole('img', { name: 'a' })).toBeDefined();
  });
  it('reacts to mouseEnter events', () => {
    const { getByTestId } = render(
      <Icon type='foo' ariaLabel='a' onMouseEnter={mIn} onMouseLeave={mOut} />
    );
    const icon = getByTestId('icon');

    userEvent.hover(icon);
    expect(mIn).toHaveBeenCalledTimes(1);
  });
  it('reacts to mouseLeave events', () => {
    const { getByTestId } = render(
      <Icon type='foo' ariaLabel='a' onMouseEnter={mIn} onMouseLeave={mOut} />
    );
    const icon = getByTestId('icon');

    userEvent.unhover(icon);
    expect(mOut).toHaveBeenCalledTimes(1);
  });
});
