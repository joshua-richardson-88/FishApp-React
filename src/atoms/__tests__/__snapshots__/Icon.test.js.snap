// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms - Icon Test matches the snapshot 1`] = `
<DocumentFragment>
  <span
    aria-label="a"
    class="fish-icon"
    data-testid="icon"
    role="img"
  >
    foo
  </span>
</DocumentFragment>
`;
